### PR TITLE
feat(alerts): Mute metric alerts for integrations

### LINKED ISSUE #50580: Add check to integrations for rule snooze
We want to snooze integration notifications if they are muted for everyone since for integrations "mute for me" doesn't make much sense. "mute for me" won't be offered to the users as an option on the front end, but we should still check that the alert is being muted for everyone in these integration checks. Tests are very important here too to confirm that we are correctly muting the integration notifications. 

Similar PR from V1 of Mute Alerts: https://github.com/getsentry/sentry/pull/49375

### PR DESCRIPTION
**Pull request by ceorourke**
A follow up to https://github.com/getsentry/sentry/pull/50986 that checks the `RuleSnooze` table before firing notifications to integrations.

Closes #50580

### CODE REVIEW

**Review by wedamija — APPROVED** [review]

**Review by roggenkemper — APPROVED** [review]

File: src/sentry/incidents/action_handlers.py [hunk 1]
```diff
@@ -47,10 +48,12 @@ def resolve(self, metric_value: int | float, new_status: IncidentStatus):
...
 class DefaultActionHandler(ActionHandler):
     def fire(self, metric_value: int | float, new_status: IncidentStatus):
-        self.send_alert(metric_value, new_status)
+        if not RuleSnooze.objects.filter(alert_rule=self.incident.alert_rule).exists():
+            self.send_alert(metric_value, new_status)
 
     def resolve(self, metric_value: int | float, new_status: IncidentStatus):
-        self.send_alert(metric_value, new_status)
+        if not RuleSnooze.objects.filter(alert_rule=self.incident.alert_rule).exists():
+            self.send_alert(metric_value, new_status)
```
Comment (wedamija) [inline src/sentry/incidents/action_handlers.py:56]: Actually should this be filtered to `RuleSnooze` rows that are only not for specific users?

**Review by wedamija — CHANGES_REQUESTED** [review]

File: src/sentry/incidents/action_handlers.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/incidents/action_handlers.py:56]: oh shit yeah

File: src/sentry/incidents/action_handlers.py [hunk 2]
```diff
@@ -48,11 +48,15 @@ def resolve(self, metric_value: int | float, new_status: IncidentStatus):
...
+        if not RuleSnooze.objects.filter(
+            alert_rule=self.incident.alert_rule, user_id__isnull=True
+        ).exists():
             self.send_alert(metric_value, new_status)
 
     def resolve(self, metric_value: int | float, new_status: IncidentStatus):
-        if not RuleSnooze.objects.filter(alert_rule=self.incident.alert_rule).exists():
+        if not RuleSnooze.objects.filter(
+            alert_rule=self.incident.alert_rule, user_id__isnull=True
+        ).exists():
```
Comment (wedamija) [inline src/sentry/incidents/action_handlers.py:59]: Not sure if we need to handle this in this pr, but I'm starting to wonder if we should have something on the manager for `RuleSnooze` that makes these queries for us.

Like `RuleSnooze.objects.disabled_for_team(alert_rule)` and `RuleSnooze.objects.disabled_for_user(alert_rule, user)` or something

**Review by wedamija — APPROVED** [review]

File: src/sentry/incidents/action_handlers.py (hunk 2 above)
Comment (ceorourke) [reply src/sentry/incidents/action_handlers.py:59]: Yeah that could definitely be useful, we can do it in a follow up. I've been meaning to make a test fixture thing for muting an alert too.

**Pull request merged**